tests/modules/images/webp-uploads/load-tests.php

```diff
@@ -572,94 +572,23 @@ function () {
-
-		// Mock attachment meta data to test when jpeg image is smaller.
-		add_filter(
-			'wp_get_attachment_metadata',
-			function( $data, $attachment_id ) {
-				$data['sources'] = array(
-					'image/jpeg' => array(
-						'file'     => 'paint.jpeg',
-						'filesize' => 1000,
-					),
-					'image/webp' => array(
-						'file'     => 'paint.webp',
-						'filesize' => 2000,
-					),
-				);
-
-				return $data;
-			},
-			10,
-			2
-		);
-
-		$mime_types = webp_uploads_get_mime_types_by_filesize( array( 'image/jpeg', 'image/webp' ), $attachment_id, 'the_content' );
-
-		$this->assertIsArray( $mime_types );
-		$this->assertSame( array( 'image/jpeg', 'image/webp' ), $mime_types );
-	}
-
-	/**
-	 * Test webp_uploads_get_mime_types_by_filesize removes invalid mime types with zero filesize.
+	 * Allow the upload of a WebP image if at least one editor supports the format
 	 *
 	 * @test
 	 */
-	public function it_should_remove_mime_types_with_zero_filesize() {
-		// File should generate smallest jpeg image size.
-		$attachment_id = $this->factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/paint.jpeg' );
+	public function it_should_use_smaller_jpg_image_when_smaller_size_preferred() {
+		// Prefer smaller image size.
+		add_filter( 'webp_uploads_prefer_smaller_image_file', '__return_true' );
 
-		// Mock attachment meta data to test mime type with zero filesize.
-		add_filter(
-			'wp_get_attachment_metadata',
-			function( $data, $attachment_id ) {
-				$data['sources'] = array(
-					'image/jpeg'    => array(
-						'file'     => 'paint.jpeg',
-						'filesize' => 1000,
-					),
-					'image/webp'    => array(
-						'file'     => 'paint.webp',
-						'filesize' => 2000,
-					),
-					'image/invalid' => array(
-						'file'     => 'paint.avif',
-						'filesize' => 0,
-					),
-				);
-
-				return $data;
-			},
-			10,
-			2
-		);
+		// File produces larger webp image, original jpg is smaller.
+		$attachment_id = $this->factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg' );
+		$metadata      = wp_get_attachment_metadata( $attachment_id );
+		$tag           = wp_get_attachment_image( $attachment_id, 'full', false, array( 'class' => "wp-image-{$attachment_id}" ) );
+		$updated_tag   = webp_uploads_img_tag_update_mime_type( $tag, 'the_content', $attachment_id );
 
-		$mime_types = webp_uploads_get_mime_types_by_filesize( array( 'image/jpeg', 'image/webp' ), $attachment_id, 'the_content' );
+		// The full size WebP is smaller.
+		$expected_tag = str_replace( $metadata['sources']['image/jpeg']['file'], $metadata['sources']['image/webp']['file'], $tag );
 
-		$this->assertIsArray( $mime_types );
-		$this->assertNotContains( 'image/invalid', $mime_types );
-		$this->assertSame( array( 'image/jpeg', 'image/webp' ), $mime_types );
+		$this->assertSame( $expected_tag, $updated_tag );
```

Comment:
Is this test in fact doing what the comment above says? It says this test covers the case where JPEG is smaller than WebP, but then for the `$expected_tag` it replaces the JPEG version with WebP, so that seems not right. If the JPEG version is smaller, the test should expect that the JPEG version is still used even though a WebP version exists.

Reply:
Ah, thanks for clarifying. That makes sense, in that case indeed the updated comments look better.